tests/integration/profiles_management/test_create_or_update_profiles.py

```diff
@@ -188,15 +216,64 @@ def test_rolebindings_are_created(lightkube_client: Client):
         resources={},
     )
 
-    log.info("Creating RoleBinding for admin role.")
-    create_or_update_profiles(lightkube_client, ProfilesManagementRepresentation([pmr_profile]))
+    log.info("Creating resources for admin role.")
+    create_or_update_profiles(
+        lightkube_client,
+        ProfilesManagementRepresentation([pmr_profile]),
+        KFP_PRINCIPAL,
+        ISTIO_PRINCIPAL,
+    )
 
     rbs = kfam.list_contributor_rolebindings(lightkube_client, ns)
     assert len(rbs) == 1
-
     assert rbs[0].metadata is not None
     assert rbs[0].metadata.annotations is not None
     assert rbs[0].metadata.annotations["user"] == user
     assert rbs[0].metadata.annotations["role"] == role
 
+    aps = kfam.list_contributor_authorization_policies(lightkube_client, ns)
+    assert len(aps) == 1
+    assert aps[0].metadata is not None
+    assert aps[0].metadata.annotations is not None
+    assert aps[0].metadata.annotations["user"] == user
+    assert aps[0].metadata.annotations["role"] == role
+
+    profiles.remove_profile(profile, lightkube_client)
+
+
+def test_authorization_policies_with_incorrect_principals_are_updated(lightkube_client: Client):
+    ns = "test-authz-policies-with-incorrect-principals-are-updated"
+    profile = profiles.apply_profile_and_resources(
+        lightkube_client, profile_path=PROFILE_PATH, resources_path=RESOURCES_PATH, namespace=ns
+    )
+
+    # Same user/role as defined in YAML files.
+    # Only change will be the principal when calling create_or_update_profiles()
+    user = "kimonas@canonical.com"
+    role = ContributorRole.EDIT
+    pmr_profile = Profile(
+        name=ns,
+        owner=Owner(name=ns, kind=UserKind.USER),
+        contributors=[Contributor(name=user, role=role)],
+        resources={},
+    )
+
+    kfp_principal = "kfp-principal"
+    istio_principal = "istio-principal"
```

Comment:
suggestion: Maybe explicitly state that the principals used here are not the same
```suggestion
    kfp_principal = "different-kfp-principal"
    istio_principal = "different-istio-principal"
```